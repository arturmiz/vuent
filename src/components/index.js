import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
