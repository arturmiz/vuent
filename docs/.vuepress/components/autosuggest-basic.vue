<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
