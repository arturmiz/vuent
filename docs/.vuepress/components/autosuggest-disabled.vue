<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
