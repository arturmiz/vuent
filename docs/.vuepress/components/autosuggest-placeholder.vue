<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
