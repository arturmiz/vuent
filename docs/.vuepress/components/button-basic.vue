<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
