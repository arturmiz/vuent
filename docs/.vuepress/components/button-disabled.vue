<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
