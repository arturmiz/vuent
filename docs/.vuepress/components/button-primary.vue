<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
