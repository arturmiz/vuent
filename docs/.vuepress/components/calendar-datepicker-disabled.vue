<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
