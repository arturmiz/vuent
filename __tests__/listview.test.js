import { createLocalVue, shallowMount } from '@vue/test-utils';
import { VntListView } from '@/components';
import { isInstalled } from './utils';

describe('ListView', () => {
  let localVue, wrapper;

  beforeAll(() => {
    localVue = createLocalVue();
    localVue.use(VntListView);
  });

  test('can be installed separately', () => {
    expect(isInstalled(localVue, VntListView)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = shallowMount(VntListView);
    });

    test('selectionMode is none', () => {
      expect(wrapper.vm.selectionMode).toBe('none');
    });

  });

  test('renders correctly simple static list', () => {
    wrapper = shallowMount(VntListView, {
      localVue,
      slots: {
        default: `
         <vnt-listview-item>
          Banana
         </vnt-listview-item>
         <vnt-listview-item>
          Apple
         </vnt-listview-item>
         <vnt-listview-item>
          Orange
         </vnt-listview-item>
        `
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('renders correctly single selection list', () => {
    wrapper = shallowMount(VntListView, {
      localVue,
      propsData: {
        selectionMode: 'single',
      },
      slots: {
        default: `
         <vnt-listview-item>
          Banana
         </vnt-listview-item>
         <vnt-listview-item>
          Apple
         </vnt-listview-item>
         <vnt-listview-item>
          Orange
         </vnt-listview-item>
        `
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
