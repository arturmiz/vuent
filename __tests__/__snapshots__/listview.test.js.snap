// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView renders correctly simple static list 1`] = `
<div role="list" class="vnt-listview">
  <div role="listitem" class="vnt-listview-item">
    Banana
  </div>
  <div role="listitem" class="vnt-listview-item">
    Apple
  </div>
  <div role="listitem" class="vnt-listview-item">
    Orange
  </div>
</div>
`;
